tests/comments/test_async_serializer.py

```diff
@@ -0,0 +1,229 @@
+    assert comment_5['user_pk'] == -1
+    assert comment_5['user_profile_url'] == ''
+    assert comment_5['is_deleted'] is True
+    assert comment_5['is_moderator'] is False
+    assert len(comment_5['child_comments']) == 0
+    assert comment_5['comment'] == Comment.objects.get(pk=5).comment
+    assert comment_5['comment'] == ''
+    assert comment_5['object_pk'] == question.pk
+    assert comment_5['is_removed'] is True
+    assert comment_5['is_censored'] is False
+    assert comment_5['is_blocked'] is False
+    assert len(comment_5['comment_categories']) == 0
+    assert comment_5['last_discussed'] is None
+    assert comment_5['is_moderator_marked'] is False
+    assert comment_5['content_type'] == question_ct.pk
+
+    # What is the deleted status of comments that have been
+    # blocked, censored, and removed?
+    # A blocked comment should not be deleted in case it must be unblocked:
+    assert comment_3['is_blocked'] is True
+    assert comment_3['is_deleted'] is False
+
+    # A censored comment is actually is_deleted.
+    assert comment_4['is_censored'] is True
+    assert comment_4['is_deleted'] is True
+    # But does this make sense or should the content remain accessible?
+
+    # A removed comment should be is_deleted:
+    assert comment_5['is_removed'] is True
+    assert comment_5['is_deleted'] is True
+
+    # The difference between is_removed and is_censored,
+    # is that is_removed is carried out by the creator,
+    # and is_censored can only be done by a moderator:
+    assert comment_5['is_moderator'] is False
+    # But this differentiation does not occur in the serializer:
+    assert comment_4['is_moderator'] is False
+    # Because the serializer does not differentiate between
+    # moderator and creator for deleting the comment,
+    # does it actually make sense to show this
+    # differentiation in the serializer?
+
+    # Why is there an is_removed property
+    # if there is also an is_deleted property?
+
+    # Testing a comment that has been marked by a moderator
+    comment_6 = comment_data[1]
+    assert comment_6['id'] == 6
+    assert comment_6['user_name'] == ''
+    assert comment_6['user_pk'] == str(user.pk)
+    assert comment_6['user_profile_url'] == ''
+    assert comment_6['is_deleted'] is False
+    assert comment_6['is_moderator'] is False
+    assert len(comment_6['child_comments']) == 0
+    assert comment_6['comment'] == Comment.objects.get(pk=6).comment
+    assert comment_6['object_pk'] == question.pk
+    assert comment_6['is_removed'] is False
+    assert comment_6['is_censored'] is False
+    assert comment_6['is_blocked'] is False
+    assert len(comment_6['comment_categories']) == 0
+    assert comment_6['last_discussed'] is None
+    assert comment_6['is_moderator_marked'] is True
+    assert comment_6['content_type'] == question_ct.pk
+    # Moderator-marked comments no longer show the user name
+    # nor the link to the user profile. Does this make sense?
```

Comment:
That's not true, is it? The moderator marked does not change anything apart from the boolean field is_moderator_marked, which is how it should be. The username and the link do not work for all comments in these tests. The link because Django's user model does not include the get_absolute_url and the name I commented above.